fairlearn/metrics/_group_metric_result.py

```diff
@@ -91,6 +99,24 @@ def range_ratio(self):
     def range_ratio(self, value):
         self._range_ratio = value
 
+    def to_dict(self, include_overall=True, include_by_group=True, include_derived=False):
+        """Return a dictionary representation of this object."""
+        result = dict()
+        if include_overall:
+            result[_OVERALL] = copy.deepcopy(self.overall)
+        if include_by_group:
+            result[_BY_GROUP] = copy.deepcopy(self.by_group)
+        if include_derived:
+            result[_MAX] = copy.deepcopy(self.maximum)
+            result[_MIN] = copy.deepcopy(self.minimum)
+            result[_RANGE] = copy.deepcopy(self.range)
+            result[_RANGE_RATIO] = copy.deepcopy(self.range_ratio)
+        return result
+
+    def __repr__(self):
+        """Generate string representation of a `GroupMetricResult`."""
+        return repr(self.to_dict())
```

Comment:
would that be equal to `self.__dict__` ?

Reply:
Depends on exactly what that does, if you look at the arguments to `to_dict()`